### PR TITLE
feat: add io/loaders module 

### PR DESCRIPTION
**Pull request by strixy16**
Includes features, general, and images specific loading functions

## Summary by CodeRabbit

- **New Features**
	- Introduced a new module for loading various data types in the READII pipeline.
	- Added functions for loading imaging feature sets, dataset configurations, and data files into DataFrames.
	- New configuration files for the 4D-Lung and NSCLC_Radiogenomics datasets created.

- **Bug Fixes**
	- Enhanced error handling for file loading functions.

- **Tests**
	- Added unit tests for configuration loading and feature extraction validation.
	- Updated existing tests for clarity and specificity.
	- Introduced new fixtures to support testing.

- **Chores**
	- Updated `.gitignore` to exclude test output files.

### CODE REVIEW

File: src/readii/io/loaders/images.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+from pathlib import Path
+from typing import Union
+
+def getImageTypesFromDirectory(raw_data_dir:Union[Path|str],
+                               feature_file_prefix:str = "",
+                               feature_file_suffix:str = ".csv"):
```
[the remark this answers is not in the document]
Comment (jjjermiah) [reply src/readii/io/loaders/images.py:6]: we are only python 3.10+

File: tests/4D-Lung/4D-Lung.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Config file for 4D-Lung for READII
+dataset_name: 4D-Lung 
+
+### CLINICAL VARIABLE INFORMATION ###
+# Event values should be in the order [Alive_value, Dead_value]
+outcome_variables:
+    time_label: 
+    event_label: 
+    convert_to_years: False
+    event_value_mapping:
```
[the remark this answers is not in the document]
Comment (jjjermiah) [reply tests/4D-Lung/4D-Lung.yaml:10]: not an issue

File: tests/4D-Lung/4D-Lung.yaml
Comment (jjjermiah) [inline tests/4D-Lung/4D-Lung.yaml:1]: in 2.0 we should discuss the schema for the settings, I assume all of these except the `image_types` are only relevant to the **`analyze`** part of the pipeline

**Review by jjjermiah — CHANGES_REQUESTED** [review]
solid, have a PR at #87 for with some minor changes for one of the files and a couple changes in the feature file

generally try to use pathlib over os.path to make reading easier

File: tests/4D-Lung/4D-Lung.yaml
Comment (strixy16) [reply tests/4D-Lung/4D-Lung.yaml:1]: yeah these can definitely be updated, I just made this for the testing stuff

**Review by jjjermiah — APPROVED** [review]

**Pull request merged**